src/uct/ib/configure.m4

```diff
@@ -282,11 +282,19 @@ AS_IF([test "x$with_ib" = "xyes"],
                [AC_DEFINE([HAVE_IBV_DM], 1, [Device Memory support])],
                [], [[#include <infiniband/verbs.h>]])])
 
-        # DDP support
         AS_IF([test "x$have_mlx5" = xyes], [
+           # DDP support
            AC_CHECK_DECLS([MLX5DV_CONTEXT_MASK_OOO_RECV_WRS],
                [AC_DEFINE([HAVE_OOO_RECV_WRS], 1, [Have DDP support])],
-               [], [[#include <infiniband/mlx5dv.h>]])])
+               [], [[#include <infiniband/mlx5dv.h>]])
+
+           # Direct NIC support, from IB side
+           have_direct_nic=yes
+           AC_CHECK_DECLS([mlx5dv_get_data_direct_sysfs_path,
+                           mlx5dv_reg_dmabuf_mr], [], [have_direct_nic=no],
+                          [[#include <infiniband/mlx5dv.h>]])
+           AS_IF([test "x$have_direct_nic" != xno],
+               [AC_DEFINE([HAVE_DIRECT_NIC], 1, [Have Direct NIC support])])])
```

Comment:
do we really need this HAVE_DIRECT_NIC macro? maybe can use the ones generated automatically from AC_CHECK_DECLS line above

Reply:
we can keep both since they are enabling independent flows in the code. query controls setting md->direct_nic_sys_dev and reg_mr is when registration happens. each macro can be used independently.